### PR TITLE
feat(logger): add e2e tests for logger

### PR DESCRIPTION
**Pull request by ijemmy**
## Description of your changes

Add e2e tests for logger

Currently, we only have e2e tests for metrics and tracer. This PR adds e2e tests similar to the other two modules. 

### How to verify this change
Run `AWS_PROFILE=<YOUR_PROFILE_NAME> AWS_REGION=eu-central-1 npx jest --group=e2e/logger/`

Here's the output:
```
Test Suites: 3 passed, 3 of 8 total
Tests:       13 passed, 13 total
Snapshots:   0 total
Time:        114.482 s
```

### Related issues, RFCs

https://github.com/awslabs/aws-lambda-powertools-typescript/issues/353 

### PR status

***Is this ready for review?:*** YES  
***Is it a breaking change?:*** NO

## Checklist

- [x] [My changes meet the tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
- [x] I have performed a *self-review* of my own code
- [x] I have *commented* my code where necessary, particularly in areas that should be flagged with a TODO, or hard-to-understand areas
- [x] I have made corresponding changes to the *documentation*
- [x] My changes generate *no new warnings*
- [x] The *code coverage* hasn't decreased
- [x] I have *added tests* that prove my change is effective and works
- [x] New and existing *unit tests pass* locally and in Github Actions
- [x] Any *dependent changes have been merged and published* in downstream module
- [x] The PR title follows the [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-typescript/blob/main/.github/semantic.yml#L2)

### Breaking change checklist

- [ ] I have documented the migration process
- [ ] I have added, implemented necessary warnings (if it can live side by side)

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by flochaz — CHANGES_REQUESTED** [review]
LGTM except cdk link

File: packages/logger/tests/helpers/InvocationLog.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  public static LEVEL = LEVEL;
+
+  /**
+   * Array of logs from invocation. 
+   * 
+   * The first element is START, and the last two elements are END, and REPORT.
+   * In each log, each content is separated by '\t'
+   *    [
+   *      'START RequestId: afb6de1a-48f8-4fbb-ad72-23a4f0c2924c Version: $LATEST',
+   *      '2022-01-27T16:04:39.323Z\tafb6de1a-48f8-4fbb-ad72-23a4f0c2924c\tINFO\t{"cold_start":true,"function_arn":"arn:aws:lambda:eu-central-1:561912387782:function:loggerMiddyStandardFeatures-c555a2ec-1121-4586-9c04-185ab36ea34c","function_memory_size":128,"function_name":"loggerMiddyStandardFeatures-c555a2ec-1121-4586-9c04-185ab36ea34c","function_request_id":"7f586697-238a-4c3b-9250-a5f057c1119c","level":"INFO","message":"This is an INFO log with some context and persistent key","service":"logger-e2e-testing","timestamp":"2022-01-27T16:04:39.323Z","persistentKey":"works"}',
```
Comment (dreamorosi) [inline packages/logger/tests/helpers/InvocationLog.ts:28]: Can we use the same ARNs we have [in the docs](https://awslabs.github.io/aws-lambda-powertools-typescript/latest/core/logger/#capturing-lambda-context-info)?

File: packages/logger/tests/e2e/basicFeatures.middy.test.ts [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+  describe('Logging an error object', () => {
+    it('should log additional keys and value only once', async () => {
+      const logMessages = invocationLogs[0].getFunctionLogs(LEVEL.ERROR)
+        .filter(message => message.includes(ERROR_MSG));
+      
+      expect(logMessages).toHaveLength(1);
+
+      // Each log contains:
+      // We will extract the last part
```
Comment (dreamorosi) [inline packages/logger/tests/e2e/basicFeatures.middy.test.ts:150]: Could you elaborate more on these comments? From the way it reads it seems a line is missing

**Review by dreamorosi — CHANGES_REQUESTED** [review]
Already looking good!

Just left some comments on the files, plus this one:
- Some files contain a copyright notice while others don't, the rest of the project also doesn't contain any so we should settle on one

**Review by dreamorosi — DISMISSED** [review]

**Review by dreamorosi — COMMENTED** [review]
Just ran the e2e tests [here on GitHub](https://github.com/awslabs/aws-lambda-powertools-typescript/runs/5023529706) and it failed, it seems there are issues with the stack deployment based on the errors.

**Comment by ijemmy** [comment]
@dreamorosi Logger uses `eu-central-1` for unit test. I need to CDK bootstrap the region.

I am considering changing everything to `eu-west-1`. But that will affect unit testing so I will discuss with @saragerion  first.

**Review by dreamorosi — APPROVED** [review]

**Review by flochaz — APPROVED** [review]
LGTM !

**Comment by ijemmy** [comment]
fyi, 1 check didn't pass (e2e) test.

It's a flaky test from metric, not logger. I have rerun and everything is green https://github.com/awslabs/aws-lambda-powertools-typescript/actions/runs/1846184047

**Pull request merged**